packages/@react-stately/data/src/useListData.ts

```diff
@@ -152,9 +152,28 @@ export function useListData<T>(options: ListOptions<T>): ListData<T> {
     filterText: initialFilterText
   });
 
+  const filterItems = (items: T[], filterText: string): T[] => {
+    return items.reduce((acc: T[], item: any) => {
+      if (item.children) {
```

Comment:
This is assuming something about the data structure of an item that isn't specified in the types. What if section objects use a different property name for their children or they come from a different data source entirely? I don't think useListData should be prescriptive about the objects that get passed into it.

If your items have children maybe useTreeData would be more appropriate?

Reply:
So listbox does actually support [being passed items from useTreeData](https://github.com/adobe/react-spectrum/blob/45a872803500086c1f589d5efb3e32b59a992b2c/packages/%40react-spectrum/listbox/stories/ListBox.stories.tsx#L937-L942), but you are right about useTreeData not supporting filtering out of the box. I've raised a question to the team about whether we'd prefer to commit to having nested item filter support supported useListData (which would then mean we'd need to update the rest of the useListData methods to work for nested data) or if we'd instead add filtering to useTreeData.

Reply:
Thanks for your patience. We had an internal discussion and decided we'd like to add filtering to useTreeData instead of having useListData be another tree implementation.

Would you like to have a go at that in this PR?

Reply:
@snowystinger yes, do I start a new PR for this?

Reply:
either one, whatever you're more comfortable with. If you use this one, we'll just rename it